crypto/signatures.go

```diff
@@ -30,11 +31,30 @@ func GenerateSignatureKeys() (sk SecretKey, pk PublicKey, err error) {
 
 // SignBytes signs a message using a secret key.
 func SignBytes(data []byte, sk SecretKey) (sig Signature, err error) {
+	if len(data) != 32 {
+		err = errors.New("will only sign 32 byte hashes!")
+		return
+	}
```

Comment:
we should probably move the hash files to crypto, and the merkle files to merkle